core/services/offchainreporting/delegate.go

```diff
@@ -146,6 +146,19 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.Service, err
 		if err != nil {
 			return nil, err
 		}
+		if peerID == "" {
+			var p2pkeys []p2pkey.KeyV2
+			p2pkeys, err = d.keyStore.P2P().GetAll()
+			if err != nil {
+				return nil, errors.Wrap(err, "while getting all p2p keys")
+			}
+			if len(p2pkeys) == 1 {
+				logger.Warn("No P2P_PEER_ID set, defaulting to first key in database")
+				peerID = p2pkeys[0].PeerID()
+			} else {
+				return nil, errors.Errorf("multiple p2p keys found but peer ID was not set. You must specify P2P_PEER_ID if you have more than one key.")
+			}
+		}
```

Comment:
Could we abstract this logic into a function? Since it appears to be similar/identical to the changes made in peerwrapper.

Reply:
moved to `P2P().GetOrFirst()`